plugins/modules/foreman_smart_class_parameter.py

```diff
@@ -0,0 +1,248 @@
+                        desired_override_value['value'] = parameter_value_to_str(desired_override_value['value'], parameter_type)
+                    current_override_value = current_override_values.pop(match, None)
+                    if current_override_value:
+                        current_override_value['value'] = parameter_value_to_str(current_override_value['value'], parameter_type)
+                    self.ensure_entity(
+                        'override_values', desired_override_value, current_override_value,
+                        state="present", entity_spec=override_value_entity_spec, params=scope)
+                for current_override_value in current_override_values.values():
+                    self.ensure_entity(
+                        'override_values', None, current_override_value, state="absent", entity_spec=override_value_entity_spec, params=scope)
+
+
+def main():
+    module = ForemanSmartClassParameterModule(
+        argument_spec=dict(
+            puppetclass_name=dict(required=True),
+            parameter=dict(required=True),
+        ),
+        entity_spec=dict(
+            parameter_type=dict(choices=['string', 'boolean', 'integer', 'real', 'array', 'hash', 'yaml', 'json', '']),
+            validator_type=dict(choices=['list', 'regexp']),
+            validator_rule=dict(),
+            description=dict(),
+            default_value=dict(type='raw'),
+            omit=dict(type='bool'),
+            override=dict(type='bool'),
+            merge_default=dict(type='bool'),
+            merge_overrides=dict(type='bool'),
+            avoid_duplicates=dict(type='bool'),
+            required=dict(type='bool'),
+            hidden_value=dict(type='bool'),
+            override_value_order=dict(type='list', elements='str'),
+            # tried nested_list here but, if using nested_list, override_values are not part of loaded entity.
+            # override_values=dict(type='nested_list', elements='dict', entity_spec=override_value_entity_spec),
+            override_values=dict(type='list', elements='dict'),
+            state=dict(default='present', choices=['present_with_defaults', 'present']),
+        ),
+        # smart_class_parameters are created on puppetclass import and cannot be created/deleted from API,
+        # so if we don't find it, it's an error.
+        entity_opts=dict(failsafe=False),
+    )
+
+    entity_dict = module.clean_params()
+    search = "puppetclass_name={0} and parameter={1}".format(entity_dict['puppetclass_name'], entity_dict['parameter'])
+    override_values = entity_dict.pop('override_values', None)
+    if 'override_value_order' in entity_dict:
+        entity_dict['override_value_order'] = '\n'.join(entity_dict['override_value_order'])
+    with module.api_connection():
+        entity, entity_dict = module.resolve_entities(search=search, entity_dict=entity_dict)
+        # Foreman API returns 'hidden_value?' instead of 'hidden_value' this is a bug ?
```

Comment:
This seems intentional.

Maybe @orrabin remembers? The code was added so long ago:

```
commit 9643a2cb1e67f4f5fb75e47e83729fa9c168fdb8
Author: Ori Rabin <orabin@redhat.com>
Date:   Sun Dec 11 15:18:49 2016 +0200

    Fixes #17653 - Allow all parameters api to unhide values
```

Reply:
If I remember correctly `hidden_value` was actually the value when hidden (so ********) but `hidden_value?` is meant to understand if you want to see it as hidden or plain text.
I removed the `hidden_value` and in stead added this partial: https://github.com/theforeman/foreman/blob/develop/app/views/api/v2/common/show_hidden.json.rabl that would show the real or hidden value in the same field.